larq_compute_engine/core/bgemm_kernels_arm.h

```diff
@@ -97,80 +66,47 @@ struct BgemmKernel<ruy::Path::kNeon, std::int8_t,
 // A BGEMM kernel for ARM64 Neon.
 #include "bgemm_kernels_arm64.h"
 
-// Optimised Aarch64 kernel for float output with 16-bit accumulators.
-template <>
-struct BgemmKernel<ruy::Path::kNeon, float,
-                   BinaryMulParams<std::int16_t, float>> {
-  Tuning tuning = Tuning::kAuto;
-  using LhsLayout = FixedKernelLayout<Order::kColMajor, 4, 8>;
-  using RhsLayout = FixedKernelLayout<Order::kColMajor, 4, 4>;
-  explicit BgemmKernel(Tuning tuning_) : tuning(tuning_) {}
-  void Run(const ruy::PMat<TBitpacked>& lhs, const ruy::PMat<TBitpacked>& rhs,
-           const BinaryMulParams<std::int16_t, float>& mul_params,
-           int start_row, int start_col, int end_row, int end_col,
-           ruy::Mat<float>* dst) const {
-    BinaryKernelParams<float, LhsLayout::kCols, RhsLayout::kCols> params;
-    MakeBinaryKernelParams(lhs, rhs, mul_params, start_row, start_col, end_row,
-                           end_col, dst, &params);
-    BinaryKernelNeonOutOfOrder8x4(params);
-  }
-};
-
-// Optimised Aarch64 kernel for int8 output with 16-bit accumulators.
-template <>
-struct BgemmKernel<ruy::Path::kNeon, std::int8_t,
-                   BinaryMulParams<std::int16_t, std::int8_t>> {
+// Optimised Aarch64 kernel with 16-bit accumulators. Supports float or int8 or
+// bitpacked output.
+template <typename DstScalar>
+struct BgemmKernel<ruy::Path::kNeon, DstScalar,
+                   BinaryMulParams<std::int16_t, DstScalar>> {
```

Comment:
The fact that this now supports all output types means it can be generic over `DstScalar`, and we can get rid of the ugliness of having to declare each kernel multiple times for each output type.